File: gulpfile.js
var gulp = require("gulp");
var ts = require("gulp-typescript");
const tsProject = ts.createProject("./tsconfig.json");
const path = {
  controller : "src/controllers",
  routes: "src/routes"
}

//console.log(dataOnel.dataTwo)

gulp.task("default", function () {
  return tsProject.src().pipe(tsProject()).js.pipe(gulp.dest(path.controller));
});

/*gulp.task("controllers", function () {
  return tsProjectTwo.src().pipe(tsProjectTwo()).js.pipe(gulp.dest("src/controllers"));
});*/

File: gulpfile.py
from invoke import task
import shutil

path = {
    'controller': 'src/controllers',
    'routes': 'src/routes'
}

@task
def default(ctx):
    build_typescript(ctx)

def build_typescript(ctx):
    print("Compiling TypeScript...")
    shutil.rmtree(path['controller'], ignore_errors=True)
    ctx.run("tsc")
    print("TypeScript compilation complete.")
